### ISSUE TITLE
Can't use mixed colors

### ISSUE THREAD

**Issue by kpym** [ho-tex/soul#2]
For some reason I can't use directly colors like `yellow!35!white`. 
The following code works
```latex
\colorlet{hl}{yellow!35!white}
\sethlcolor{hl}
```
but the following one don't
```latex
\sethlcolor{yellow!35!white}
```
It looks like that `soul(utf8)` use under the hood `\textcolor` which is ok with mixed colors. So I can't get it why it is not working.

**Comment by u-fischer** [comment]
Please show a small but complete example. That makes it much easier to test the issue.

**Comment by kpym** [comment]
@u-fischer Here is an example of MWE

```latex
\documentclass{article}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[utf8]{inputenc}
\else
  \usepackage{fontspec}
\fi
% --------------------------
\usepackage{xcolor}
\colorlet{hl}{red!35!white}
% --------------------------
\usepackage{soulutf8}
% --------------------------
\begin{document}
  \sethlcolor{hl}
  \hl{Test using \texttt{hl} color}

  \sethlcolor{red!35!white}
  \hl{Test using \texttt{red!35!white} color}
\end{document}
```
![image](https://user-images.githubusercontent.com/3142701/115957537-52f1df00-a503-11eb-8ed9-5b0526bba5eb.png)

**Comment by kpym** [comment]
@u-fischer This behavior is not specific to `soulutf8` because we can see the same thing with the original `soul` package.

```latex
\documentclass{article}
% --------------------------
\usepackage{xcolor}
\colorlet{hl}{blue!35!white}
% --------------------------
\usepackage{soul}
% --------------------------
\begin{document}
  \sethlcolor{hl}
  \hl{Test using \texttt{hl} color}

  \sethlcolor{blue!35!white}
  \hl{Test using \texttt{blue!35!white} color}
\end{document}
```
![image](https://user-images.githubusercontent.com/3142701/115957703-4752e800-a504-11eb-98d8-349a77783e83.png)

**Comment by davidcarlisle** [comment]
a quick workaround would seem to be

```
\documentclass{article}
% --------------------------
\usepackage{xcolor}
\colorlet{hl}{blue!35!white}
% --------------------------
\usepackage{soul}
\let\zz\sethlcolor
\def\sethlcolor#1{\colorlet{soul-#1}{#1}\zz{soul-#1}}
% --------------------------
\begin{document}
  \sethlcolor{hl}
  \hl{Test using \texttt{hl} color}

  \sethlcolor{blue!35!white}
  \hl{Test using \texttt{blue!35!white} color}
\end{document}
```

![image](https://user-images.githubusercontent.com/1268738/115957805-68afd600-a4fc-11eb-9c53-9cb4ef62c555.png)

**Comment by kpym** [comment]
@davidcarlisle We can even do a more dirty thing : `\def\sethlcolor#1{\colorlet{#1}{#1}\zz{#1}}` ;)

**Comment by kpym** [comment]
What I do not understand is that in the code it looks like `\sethlcolor` use the color only through `\textcolor{#1}` which should be ok... but probably I miss something.

**Comment by u-fischer** [comment]
@kpym  It is a known incompability between soul and xcolor, see https://tex.stackexchange.com/a/520315/2388 

We will look for a real fix.

**Comment by ktzanev** [comment]
@u-fischer Tanks for considering my request and to pointing me to the TeX.SX topic.

**Comment by kpym** [comment]
@u-fischer Tanks for considering my request and to pointing me to the TeX.SX topic.

**Comment by u-fischer** [comment]
This has been resolved with version 3.0.
